sdk/python/kfp/dsl/_container_op.py

```diff
@@ -12,309 +12,940 @@
-
-    Args:
-      cpu: A string which can be a number or a number followed by "m", which means 1/1000.
     """
-
-    self._validate_cpu_string(cpu)
-    return self.add_resource_limit("cpu", cpu)
-
-  def set_gpu_limit(self, gpu, vendor = "nvidia"):
-    """Set gpu limit for the operator. This function add '<vendor>.com/gpu' into resource limit. 
-    Note that there is no need to add GPU request. GPUs are only supposed to be specified in 
-    the limits section. See https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/.
-
-    Args:
-      gpu: A string which must be a positive number.
-      vendor: Optional. A string which is the vendor of the requested gpu. The supported values 
-        are: 'nvidia' (default), and 'amd'. 
+    Attributes:
+      swagger_types (dict): The key is attribute name
+                            and the value is attribute type.
+      attribute_map (dict): The key is attribute name
+                            and the value is json key in definition.
     """
+    # remove `name` from swagger_types so `name` is not generated in the JSON
+    swagger_types = {
+        key: value
+        for key, value in V1Container.swagger_types.items() if key != 'name'
+    }
+    attribute_map = {
+        key: value
+        for key, value in V1Container.attribute_map.items() if key != 'name'
+    }
+
+    def __init__(self, image: str, command: List[str], args: List[str],
+                 **kwargs):
+        """Creates a new instance of `Container`.
+        
+        Args:
+            image {str}: image to use, e.g. busybox:latest
+            command {List[str]}: entrypoint array.  Not executed within a shell.
+            args {List[str]}: arguments to entrypoint.
+            **kwargs: keyword arguments for `V1Container`
+        """
+        # set name to '' if name is not provided
+        # k8s container MUST have a name
+        # argo workflow template does not need a name for container def
+        if not kwargs.get('name'):
+            kwargs['name'] = ''
+
+        super(Container, self).__init__(
+            image=image, command=command, args=args, **kwargs)
+
+    def _validate_memory_string(self, memory_string):
+        """Validate a given string is valid for memory request or limit."""
+
+        if isinstance(memory_string, _pipeline_param.PipelineParam):
+            if memory_string.value:
+                memory_string = memory_string.value
+            else:
+                return
+
+        if re.match(r'^[0-9]+(E|Ei|P|Pi|T|Ti|G|Gi|M|Mi|K|Ki){0,1}$',
+                    memory_string) is None:
+            raise ValueError(
+                'Invalid memory string. Should be an integer, or integer followed '
+                'by one of "E|Ei|P|Pi|T|Ti|G|Gi|M|Mi|K|Ki"')
+
+    def _validate_cpu_string(self, cpu_string):
+        "Validate a given string is valid for cpu request or limit."
+
+        if isinstance(cpu_string, _pipeline_param.PipelineParam):
```

Comment:
Could we extract this block of code as a separate internal function because it has been used by many functions here?